packages/@aws-cdk/aws-stepfunctions-tasks/lib/codebuild/start-build.ts

```diff
@@ -12,12 +12,187 @@ export interface CodeBuildStartBuildProps extends sfn.TaskStateBaseProps {
    * CodeBuild project to start
    */
   readonly project: codebuild.IProject;
+  /**
+   * Build output artifact settings that override, for this build only.
+   *
+   * @default - the latest ones already defined in the build project.
+   */
+  readonly artifactsOverride?: codebuild.IArtifacts;
+  /**
+   * A buildspec file declaration that overrides, for this build only.
+   *
+   * @default - the latest one already defined in the build project.
+   */
+  readonly buildspecOverride?: string;
+  /**
+   * Contains information that defines how the build project reports the build status to the source provider.
+   *
+   * @default - the latest one already defined in the build project only if the build's source is GitHub, GitHub Enterprise, or Bitbucket.
+   */
+  readonly buildStatusConfigOverride?: codebuild.CfnProject.BuildStatusConfigProperty;
```

Comment:
we should not leak out `Cfn` types through public APIs - I'm guessing this is not modeled in the `aws-codebuild` module yet?

if there isn't a type that can be reused, we should model it as a type